client/cgo/verification.go

```diff
@@ -112,8 +142,15 @@ func cgoVerifyAuthPath(treeHash unsafe.Pointer, treeHashSize C.int,
 	ap.TreeNonce = tn
 	ap.LookupIndex = li
 	ap.PrunedTree = pt
+	ap.Leaf = &leafNode{
```

Comment:
> Kind of doubled down on the nils here, because doing this makes the signature to the verification functions a lot cleaner, and clearer.

Can you please elaborate on this? I might be missing something here, but I'm not sure how setting the `leafNode`'s `value` and `salt` to `nil` make the signature to the verification functions easier. Also, these two values would only be `nil` in only some cases, no?